code/datums/components/religious_tool.dm

```diff
@@ -0,0 +1,135 @@
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Enables access to the global sect directly
+	var/datum/religion_sect/easy_access_sect
+	/// What extent do we want this religious tool to act? In case you don't want full access to the list. Generated on New
+	var/operation_flags
+	/// The rite currently being invoked
+	var/datum/religion_rites/performing_rite
+
+/datum/component/religious_tool/Initialize(_flags = ALL)
+	SetGlobalToLocal() //attempt to connect on start in case one already exists!
+	operation_flags = _flags
+
+/datum/component/religious_tool/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_RELIGION_NEWSECT, .proc/SetGlobalToLocal)
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY,.proc/AttemptActions)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/component/religious_tool/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_PARENT_ATTACKBY,COMSIG_RELIGION_NEWSECT))
+
+/**
+  * Sets the easy access variable to the global if it exists.
+  */
+/datum/component/religious_tool/proc/SetGlobalToLocal()
+	if(!GLOB.religious_sect)
+		return
+	easy_access_sect = GLOB.religious_sect
+
+
+/**
+  * Since all of these involve attackby, we require mega proc. Handles Invocation, Sacrificing, And Selection of Sects.
+  */
+/datum/component/religious_tool/proc/AttemptActions(datum/source, obj/item/the_item, mob/living/user)
+	if(!istype(the_item, /obj/item/storage/book/bible)) //if it's not a bible we are going to try to sacrifice.
+		if(operation_flags & RELIGION_TOOL_SACRIFICE)
+			if(!(easy_access_sect?.can_sacrifice(the_item,user)))
+				return
+			easy_access_sect.on_sacrifice(the_item,user)
+			return COMPONENT_NO_AFTERATTACK
+	else if(easy_access_sect) //we have a bible so let's see if we have a sect!
+		if(!(operation_flags & RELIGION_TOOL_INVOKE))
+			return
+		// At this point you're intentionally trying to invoke rites, so we dont need to afterattack.
+		. = COMPONENT_NO_AFTERATTACK
+		if(!easy_access_sect.rites_list)
+			to_chat(user, "<span class='notice'>Your sect doesn't have any rites to perform!")
+			return
+		if(performing_rite)
+			to_chat(user, "<span class='notice'>There is a rite currently being performed here already!")
+			return
+		var/rite_select = input(user,"Select a rite to perform!","Select a rite",null) in easy_access_sect.rites_list
+		if(!rite_select || !user.canUseTopic(parent, BE_CLOSE, FALSE, NO_TK))
+			to_chat(user,"<span class ='warning'>You cannot perform the rite at this time.</span>")
+			return
+		var/selection2type = easy_access_sect.rites_list[rite_select]
+		performing_rite = new selection2type(parent)
+		if(!performing_rite.perform_rite(user, parent))
+			QDEL_NULL(performing_rite)
+		else
+			performing_rite.invoke_effect(user, parent)
+			easy_access_sect.adjust_favor(-performing_rite.favor_cost)
+			QDEL_NULL(performing_rite)
+		return
+	else if((operation_flags & RELIGION_TOOL_SECTSELECT)) //No bible, No sect, time to select a new one!
```

Comment:
performance wise it makes sense to do this check last but it also results in you checking if easy access sect is set a ton in the other procs, so It'd be better to just check easy access sect on top then check if this is a sectselect tool

Reply:
also those double parenthesis are pretty epic